frontend/packages/pipelines-plugin/src/components/pipelineruns/triggered-by/ResourceKebab.tsx

```diff
@@ -0,0 +1,38 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { useTranslation } from 'react-i18next';
+import { Kebab, ResourceKebabProps } from '@console/internal/components/utils';
+import { connectToModel } from '@console/internal/kinds';
+
+export const ResourceKebab = connectToModel((props: ResourceKebabProps) => {
+  const { t } = useTranslation();
+  const { actions, kindObj, resource, isDisabled, customData, terminatingTooltip } = props;
+  if (!kindObj) {
+    return null;
+  }
+  const options = _.reject(
+    actions.map((a) => a(kindObj, resource, null, customData)),
+    'hidden',
+  );
+  const isResourceLoadedFromTR =
+    resource?.metadata?.annotations?.['resource.loaded.from.tektonResults'];
+  const isResourceDeletedInK8s = resource?.metadata?.annotations?.['resource.deleted.in.k8s'];
+  return (
+    <Kebab
+      options={options}
+      key={resource.metadata.uid}
+      isDisabled={
+        isDisabled ??
+        ((_.has(resource.metadata, 'deletionTimestamp') && !isResourceLoadedFromTR) ||
+          (options.length === 1 && _.isEmpty(options[0])))
+      }
+      terminatingTooltip={
+        _.has(resource.metadata, 'deletionTimestamp') && !isResourceLoadedFromTR
+          ? terminatingTooltip || t('pipelines-plugin~Resource is being deleted.')
+          : isResourceDeletedInK8s
+          ? t('pipelines-plugin~Resource is being fetched from Tekton Results.')
+          : ''
+      }
```

Comment:
Is the delete option disabled in case PLR is deleted and fetched from the result database? But here button is disabled with no text and shows only extra white space.

<img width="1512" alt="image" src="https://github.com/openshift/console/assets/2561818/daf13ae0-ba2b-406e-b3c8-c19b8ac83ad0">

Reply:
Updated the PR to handle this.